File: Nian_Karl/QL/src/org/uva/ql/ast/expression/unary/Unary.java
package org.uva.ql.ast.expression.unary;

import org.uva.ql.ast.expression.Expression;

public abstract class Unary extends Expression {

	protected final Expression expr;
	
	public Unary(Expression expr) {
		this.expr = expr;
	}
	
	public Expression getExpression() {
		return expr;
	}

}

File: FelipezConde/QuestionnaireLanguage/AST/Nodes/Expressions/Unaries/Unary.cs
using Expressions = AST.Nodes.Interfaces;

namespace AST.Nodes.Expressions.Unaries
{
    public abstract class Unary : Expression
    {
        private readonly Expression child;

        protected Unary(Expression child, PositionInText pos)
            :base(pos)
        { this.child = child; }

        public Expression GetChildExpression()
        { return child; }

        public abstract override T Accept<T>(VisitorInterfaces.IExpressionVisitor<T> visitor);
    }
}
